specs/factorial_spec.rb

```diff
@@ -0,0 +1,15 @@
+require 'rspec'
+require_relative '../solver'
+
+RSpec.describe 'Solver' do
+  it 'Returns factorial of a positive integer' do
+    solver = Solver.new
+    factorial_result = solver.factorial(5)
+    expect(factorial_result).to eq(120)
+  end
+  it 'Returns factorial of a positive integer' do
+    solver = Solver.new
+    factorial_result = solver.factorial(4)
+    expect(factorial_result).to eq(24)
+  end
+end
```

Comment:
- [ ] Its a requirement that there should be a test case for the special case  - when input is 0, the result should equal 1. kindly add this as well
 e.g
```
expect(resolver.factorial(0)).to eql(1)
```